xmtp_mls/src/groups/device_sync.rs

```diff
@@ -101,57 +107,141 @@ pub enum DeviceSyncError {
     UnspecifiedDeviceSyncKind,
     #[error("sync reply is too old")]
     SyncReplyTimestamp,
+    #[error(transparent)]
+    Subscribe(#[from] SubscribeError),
+}
+
+impl<ApiClient, V> Client<ApiClient, V>
+where
+    ApiClient: XmtpApi + 'static,
```

Comment:
we can remove unsafe `impl Send` by requiring `Send + Sync` here, and it will work in wasm -- its the same that we do for our stream functions (look at `stream_all_messages` for as a reference). It works, because `MlsGroup` _is_ actually `Send + Sync` in wasm, its the futures that aren't

since this is a separate `impl` we avoid the headache of making `Send + Sync` in the larger `impl` block